src/lib.rs

```diff
@@ -169,7 +181,36 @@ pub enum OutputConflictHandling {
     Bless,
 }
 
-pub type Filter = Vec<(Regex, &'static [u8])>;
+/// A filter's match rule.
+#[derive(Debug)]
+pub enum Match {
+    /// If the regex matches, the filter applies
+    Regex(Regex),
+    /// If the path is found, the filter applies
```

Comment:
"path"? There is no path here. Should probably be "if that exact sequence of bytes is found"?

Reply:
ah yea, I had a `PathBuf` in there earlier